### PR TITLE
[feature] tabular error handling

### PR DESCRIPTION
**Pull request by mfraezz**
Purpose
=======
Better error handling when rendering tabular files

Changes
=======
* new exception class
* try/excepts at points where errors arising from non-ascii characters can occur

Side Effects
==========
None

### CODE REVIEW

File: mfr/ext/tabular/render.py [hunk 1]
```diff
@@ -50,12 +54,18 @@ def render_html(fp, src=None):
     table_size = 'small_table' if len(columns) < 9 else 'big_table'
     slick_grid_options = config.get('slick_grid_options').get(table_size)
 
+    try:
+        columns = json.dumps(columns)
+        rows = json.dumps(rows)
+    except UnicodeDecodeError:
+        raise UnexpectedFormattingException()
```
Comment (sloria) [inline mfr/ext/tabular/render.py]: Add a descriptive error message.

Comment (mfraezz) [inline mfr/ext/tabular/render.py]: The default error message that will be displayed is `Unable to render. Download(link) file to view it. [any additional msg goes here]`

File: mfr/ext/tabular/render.py [hunk 2]
```diff
@@ -35,7 +36,10 @@ def render_html(fp, src=None):
     :return: RenderResult object containing html and assets
     """
 
-    columns, rows = populate_data(fp)
+    try:
+        columns, rows = populate_data(fp)
+    except KeyError:
```
Comment (sloria) [inline mfr/ext/tabular/render.py]: I'm thinking this `KeyError` should be handled lower in the call stack; maybe in one of the function in `libs`, or--if that produced lots of duplication--in `populate_data`.

**Comment by sloria** [comment]
Pass finished. :ocean:

File: mfr/ext/tabular/render.py (hunk 2 above)
Comment (mfraezz) [inline mfr/ext/tabular/render.py]: Good call. The key in question tends to be a hex value (e.g. `KeyError: '\x94'`), and while in testing it I've only found KeyErrors raised during the rendering of .dta files, I don't know for sure that it's limited to them. `populate_data` seems the safer place to put it if we want it to cover all tabular filetypes without all the duplication.

**Pull request merged**